src/halmos/__main__.py

```
@@ -1357,30 +1366,19 @@ def parse_build_out(args: HalmosConfig) -> Dict:
                         contract_name,
                         sol_dirname,
                     )
-                contract_map[contract_name] = (json_out, contract_type, natspec)
 
-                try:
-                    bytecode = contract_map[contract_name][0]["bytecode"]["object"]
-                    contract_mapping_info = Mapper().get_contract_mapping_info_by_name(
-                        contract_name
-                    )
-
-                    if contract_mapping_info is None:
-                        Mapper().add_contract_mapping_info(
-                            contract_name=contract_name,
-                            bytecode=bytecode,
-                            nodes=[],
+                contract_map[contract_name] = (json_out, contract_type, natspec)
+                parse_symbols(args, contract_map, contract_name)
+
+            except KeyError as err:
+                if str(err) in ["ast", "metadata"]:
+                    # somewhat expected, e.g. for sol files containing only libraries
+                    if args.debug:
+                        debug(
+                            f"Skipped {json_filename} due to parsing failure: "
+                            "{type(err).__name__}: {err}"
                         )
-                    else:
-                        contract_mapping_info.bytecode = bytecode
-
-                    contract_mapping_info = Mapper().get_contract_mapping_info_by_name(
-                        contract_name
-                    )
-                    Mapper().parse_ast(contract_map[contract_name][0]["ast"])
 
-                except Exception:
-                    pass
```

this is what prompted the refactoring, we don't want to ignore all exceptions